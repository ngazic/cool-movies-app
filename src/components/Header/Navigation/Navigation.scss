.navigation {
  margin-bottom: 30px;

  &__link {
    display: inline-block;
    padding: 5px 10px;
    cursor: pointer;
    text-decoration: none;
    color: #4a4a4a;
    background: white;
    border: solid 2px #4a4a4a;
    border-right: none;
    font-weight: bold;
    box-shadow: 3px 5px #4a4a4a;

    &:visited {
      color: inherit;
    }

    &:last-of-type {
      border-right: solid 2px #4a4a4a;
      ;
    }
    
    &:hover {
      background-color: rgba(#4a4a4a, 0.7);
      color:white;
    }
    &.active {
      background-color: #99ccff;
    }

  }
}