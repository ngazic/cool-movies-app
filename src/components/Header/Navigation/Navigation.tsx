import React, { FunctionComponent, MouseEvent } from 'react';
import { Link, useLocation } from 'react-router-dom';
import './Navigation.scss';

interface NavigationProps {
  click: (e: MouseEvent<HTMLAnchorElement>) => void
}
const Navigation: FunctionComponent<NavigationProps> = (props) => {
  const location = useLocation();

  const setActiveLink = (path: string): string => {
   return location.pathname===`/${path}`?'active':''
  }
  
  const onClickHandler = (e: MouseEvent<HTMLAnchorElement>) => {
    props.click(e);
    e.currentTarget.parentElement?.querySelector('.active')?.classList.remove('active');
    e.currentTarget.classList.add('active');
  }

  return (
    <nav className='navigation' >
      <Link to='/movie' onClick={onClickHandler} className={"navigation__link "+setActiveLink('movie')}>
        Movies
     </Link>
      <Link to='/tv' onClick={onClickHandler} className={"navigation__link "+setActiveLink('tv')}>
        TV Shows
     </Link>
    </nav>
  );
};

export default Navigation;