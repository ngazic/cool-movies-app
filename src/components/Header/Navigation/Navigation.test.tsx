import React from "react";
import { render } from "@testing-library/react";
import Navigation from "./Navigation";
import { Router } from "react-router-dom";
import userEvent from "@testing-library/user-event";
import { createMemoryHistory } from "history";

describe("src/components/Header/Navigation/Navigation.tsx", () => {
  const history = createMemoryHistory();
  const spy = jest.spyOn(history, "push");
  const renderWithRouter = (component: React.ReactElement) => {
    return {
      ...render(<Router history={history}>{component}</Router>)
    };
  };

  it("should match snapshot", () => {
    const { asFragment } = renderWithRouter(<Navigation />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should render two navigation links", () => {
    const { getAllByRole } = renderWithRouter(<Navigation />);
    const links = getAllByRole("link");
    expect(links).toHaveLength(2);
  });

  it('clicking "movies" should redirect to "/movies"', () => {
    const { getByText } = renderWithRouter(<Navigation />);
    const moviesLink = getByText("Movies");
    userEvent.click(moviesLink);
    // expect(history.location.pathname).toBe("/hello");
    expect(spy).toBeCalledWith("/");
  });

  it('clicking "TV Shows" should redirect to "/shows"', () => {
    const { getByText } = renderWithRouter(<Navigation />);
    const moviesLink = getByText(/tv shows/i);
    userEvent.click(moviesLink);
    expect(spy).toBeCalledWith("shows");
    // screen.debug();
  });
});
