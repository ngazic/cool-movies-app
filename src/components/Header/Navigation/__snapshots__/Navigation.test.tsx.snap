// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header/Navigation/Navigation.tsx should match snapshot 1`] = `
<DocumentFragment>
  <nav
    class="navigation"
  >
    <a
      href="/movies"
    >
      Movies
    </a>
    <a
      href="/shows"
    >
      TV Shows
    </a>
  </nav>
</DocumentFragment>
`;
